packages/winpanda/extra/src/winpanda/core/cmdconf.py

```diff
@@ -354,30 +319,120 @@ def get_dcos_conf(self):
+                    f'DC/OS aggregated config package: {dcoscfg_pkg_fpath}:'
+                    f' {type(e).__name__}: {e}'
                 )
             else:
                 raise
         else:
+            LOG.debug(f'{self.msg_src}: DC/OS aggregated config package:'
+                      f' {dcoscfg_pkg_fpath}: Preprocess: OK')
             return {'template': template, 'values': values}
         finally:
             dcoscfg_pkg_fpath.unlink()
 
+    def get_dstor_dcoscfgpkg_path(self, dstor_root_url, dstor_lpi_path):
+        """Retrieve the Linux Package Index (LPI) object from the DC/OS
+        distribution storage and discover a relative URL to the DC/OS
+        aggregated configuration package.
+        LPI is expected to be a JSON-formatted file containing descriptors for
+        DC/OS distribution packages:
+
+        {
+            "<pkg-name>":{
+                "filename":"<base-path>/<pkg-name>--<pkg-version>.tar.xz",
+                "id":"<pkg-name>--<pkg-version>"
+            },
+            ...
+        }
+
+        :param dstor_root_url:         str, DC/OS distribution storage root URL
+        :param dstor_lpi_path:         str, URL path to the DC/OS Linux package
+                                       index object at the DC/OS distribution
+                                       storage
+        :return dstor_dcoscfgpkg_path: str, URL path to the DC/OS aggregated
+                                       config package at the DC/OS distribution
+                                       storage
+        """
+        dcos_conf_pkg_name = 'dcos-config-win'
+
+        # Linux package index direct URL
+        lpi_url = posixpath.join(dstor_root_url, dstor_lpi_path)
+        lpi_fname = Path(dstor_lpi_path).name
+
+        try:
+            cm_utl.download(lpi_url, str(self.inst_storage.tmp_dpath))
+            LOG.debug(f'{self.msg_src}: DC/OS Linux package index: Download:'
+                      f' {lpi_fname}: {lpi_url}')
+        except Exception as e:
+            raise cr_exc.RCDownloadError(
+                f'DC/OS Linux package index: {lpi_fname}:'
+                f' {lpi_url}: {type(e).__name__}: {e}'
+            ) from e
+
+        lpi_fpath = self.inst_storage.tmp_dpath.joinpath(lpi_fname)
+
+        try:
+            lpi = cr_utl.rc_load_json(lpi_fpath,
+                                      emheading='DC/OS Linux package index')
+
+            if not isinstance(lpi, dict):
+                raise cr_exc.RCInvalidError(
+                    f'DC/OS Linux package index: {lpi_fpath}: Invalid'
+                    f' structure: {lpi}'
+                )
+
+            dcos_conf_pkg_desc = lpi.get(dcos_conf_pkg_name)
+
+            if dcos_conf_pkg_desc is None:
+                raise cr_exc.RCElementError(
+                    f'DC/OS Linux package index: {lpi_fpath}: DC/OS aggregated'
+                    f' config package descriptor is missed:'
+                    f' {dcos_conf_pkg_name}: {lpi}'
```

Comment:
Rather than "aggregated config package descriptor", can we just have "dcos-config-win is missing from {lpi_url}"?

Reply:
The **{dcos_conf_pkg_name}** in the error message is replaced with actual key name which has been missed whatever it could be. Now it's "dcos-config-win".